subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/Dependencies.java

```diff
@@ -40,6 +43,16 @@ public interface Dependencies {
     @Inject
     DependencyFactory getDependencyFactory();
 
+    /**
+     * The current project. You need to use {@link #project()} or {@link #project(String)} to add a {@link ProjectDependency}.
+     *
+     * @return current project
+     *
+     * @since 8.0
+     */
+    @Inject
+    Project getProject();
```

Comment:
I'm not sure if I'm really a fan of having the project itself freely available on any given dependencies block. I get we need this if we want default implementations for the two methods below, but maybe we can add methods to `DependencyFactory` so we don't need to pass it in actual project instances?

Reply:
Dependency blocks in places where projects aren't a valid option is maybe a relevant edge case.  I can't think of any examples that aren't completely arbitrary though.  

I suppose in that case the custom extending interface would override these methods with a default method that throws.  I think it's probably rare enough that extracting a separate `ProjectDependencies` from this interface would be overkill.

Reply:
What I mean is that it's just odd for a `getProject()` method which returns `Project` instance to live a dependencies block at all. I'm not suggesting removing the `ProjectDependency` creating methods, but that the `Project getProject()` method seems like it doesn't really belong here.